### PR TITLE
checking zone cache integrity on RRset updates

### PR DESCRIPTION
**Pull request by m0rcq**
Merging zone cache integrity checks in master

### CODE REVIEW

File: src/erldns_zone_cache.erl [hunk 1]
```diff
@@ -288,6 +288,17 @@ put_zone_rrset({ZoneName, Digest, Records, _Keys}, RRFqdn, Type, Counter) ->
 		  lager:debug("Not processing RRSet (~p) for Zone (~p): counter (~p) provided is lower than system", [RRFqdn, ZoneName, Counter]);
 	  true -> 
 		  lager:debug("Processing RRSet (~p) for Zone (~p): counter (~p) provided is higher than system", [RRFqdn, ZoneName, Counter]),
+	          % we need to perform extra check to verify that the zone is in cache already as we might
+		  % face timing issue if the initial full zone load for subdomain has not updated the storage yet
+		  % and find_zone_in_cache() will return the parent zone instead
+		  case get_zone(erldns:normalize_name(ZoneName)) of
+			{error,zone_not_found} ->
+			     lager:debug("Zone: ~p not in cache, initiating fetch_zone", [ZoneName]),
+			     erldns_zoneclient:fetch_zone(ZoneName),
```
Comment (aeden) [inline src/erldns_zone_cache.erl:297]: `erldns_zoneclient` is not guaranteed to be present, we cannot call fetch_zone here.

**Review by aeden — CHANGES_REQUESTED** [review]

File: src/erldns_zone_cache.erl (hunk 1 above)
Comment (aeden) [reply src/erldns_zone_cache.erl:297]: It would also create a circular dependency between `erldns` and `erldns_zoneclient`

Comment (m0rcq) [reply src/erldns_zone_cache.erl:297]: understood - will refactor to address this.

**Comment by aeden** [comment]
Stale, fixed in another PR.

**Pull request closed without merging**